### PR TITLE
Renaming HandshakeType names in IANA request

### LINKED ISSUE #264: Rename TLS HandshakeTypes
This I-D registers two TLS HandshakeTypes. All of the other fields in the registry use snake_case.

### PR DESCRIPTION
**Pull request by seanturner**
Resolves #264.

### CODE REVIEW

**Review by kaduk — COMMENTED** [review]
Do we want anything in the prose about how the RequestConnectionId structure is the message contents when the HandshakeType is request_connection_id/etc.?
Also update Appendix A.2?

**Comment by seanturner** [comment]
As far as referring to the snake case in the main body, I went and looked at RFC 8446 and it does not refer to the snake case.  I tend to think that if we make the change in 5.2 and A.2, that should be enough breadcrumbs for people to figure it out.

I put them in the same order as they are in the registry and moved up new_session_ticket.

Triple check this change please.

**Review by chris-wood — APPROVED** [review]

File: draft-ietf-tls-dtls13.md [hunk 1]
```diff
@@ -1130,12 +1132,14 @@ fragmentation, DTLS modifies the TLS 1.3 handshake header:
             case client_hello:          ClientHello;
             case server_hello:          ServerHello;
             case end_of_early_data:     EndOfEarlyData;
+            case new_session_ticket:    NewSessionTicket;
             case encrypted_extensions:  EncryptedExtensions;
-            case certificate_request:   CertificateRequest;
+            case request_connection_id: RequestedConnectionId;
```
Comment (kaduk) [inline draft-ietf-tls-dtls13.md:1137]: Shouldn't this be just "RequestConnectionId", not "Requested"?

File: draft-ietf-tls-dtls13.md [hunk 2]
```diff
@@ -1130,12 +1132,14 @@ fragmentation, DTLS modifies the TLS 1.3 handshake header:
             case client_hello:          ClientHello;
             case server_hello:          ServerHello;
             case end_of_early_data:     EndOfEarlyData;
+            case new_session_ticket:    NewSessionTicket;
```
Comment (kaduk) [inline draft-ietf-tls-dtls13.md:1135]: I see NST (4) as before EOED (5) in the registry.

Comment (kaduk) [reply draft-ietf-tls-dtls13.md:1135]: Were we planning to order these by codepoint value in the final RFC?

Comment (seanturner) [reply draft-ietf-tls-dtls13.md:1135]: Yes ... will submit a new PR now to reorder

File: draft-ietf-tls-dtls13.md (hunk 1 above)
Comment (seanturner) [reply draft-ietf-tls-dtls13.md:1137]: Yes fixing this too.

**Pull request merged**